tests/integration/query/query_test.go

```diff
@@ -15,6 +15,7 @@ func TestQuery(t *testing.T) {
 		Dir:          "step1",
 		StackName:    "query-stack",
 		Dependencies: []string{"@pulumi/pulumi"},
+		CloudURL:     "file://~", // Required; we hard-code the stack name
```

Comment:
Can we instead add some randomness to the stack name and then communicate that through config? This will break on machines that happen to be using the local backend _and_ have a stack named `query-stack`?

Reply:
yes.

Reply:
Actually, this is going to get kind of complicated. `query` does not have a notion of config—so unless you have any bright ideas, I think we might have to do something really gross, like getting the stack name out of the stack's YAML file.

Reply:
Ah, yuck. Okay, let's leave the local cloud approach in for now.

Reply:
I'll file an issue to track.

Reply:
We should consider setting an envvar or something in the future. In general it seems like it'd be useful to have some way to pass args to a query.

Reply:
I'll add a bunch of nonsense to the end of the stack name—if we're going to go this direction, we can at least make it _VERY_ unlikely it will collide.